File: 143-reorder-list/143-reorder-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {void} Do not return anything, modify head in-place instead.
 */
var reorderList = function(head) {
    
    let order = [];
    let node = head;
    while(node !== null){
        order.push(node.val);
        node = node.next;
    };
    
    let left = 0;
    let right = order.length - 1;
    
    while(left <= right){
        if(left != right){
            head.val = order[left];
            head.next.val = order[right];
            right -- ;
            left ++ ;
            head = head.next.next
        }else{            
            head.val = order[left];
            return
        }
    }
    
};

File: 143-reorder-list/143-reorder-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        """
        Do not return anything, modify head in-place instead.
        """
        
        def preorder(head):
            result = []  # Initialize an empty list to store the linked nodes

            # Traverse the linked list and add each node to the result list
            while head is not None:
                result.append(head.val)
                head = head.next

            return result
        
        orde = preorder(head)
        
        left,right = 0 , len(orde)-1
        
        while left <= right:
            
            if left == right:
                head.val = orde[left]
            else:
                head.val, head.next.val = orde[left],orde[right]

                head = head.next.next
            
            left += 1
            
            right -= 1
            
        
        
            
            
        